lib/active_job/retry.rb

```diff
@@ -117,6 +125,18 @@ def check_adapter!
       end
     end
 
+    def validate_params
+      if retry_callback && !retry_callback.is_a?(Proc)
```

Comment:
Actually, maybe this should just be `retry_callback.responds_to?(:call) && retry_callback.method(:call).arity == 2`? Then it doesn't _have_ to be a proc - it can be anything "proc-like", which is probably more convenient.